src/editor.c

```diff
@@ -1523,27 +1606,58 @@ static void insert_indent_after_line(GeanyEditor *editor, gint line)
 	g_free(text);
 }
 
+static gboolean auto_close_chars_consume(ScintillaObject *sci, gint pos, gchar c) {
+	gboolean isAutoClosed = FALSE;
+	switch (c)
+	{
+		case ')':
+			if (editor_prefs.autoclose_chars & GEANY_AC_PARENTHESIS)
+				isAutoClosed = TRUE;
+			break;
+		case '}':
+			if (editor_prefs.autoclose_chars & GEANY_AC_CBRACKET)
+				isAutoClosed = TRUE;
+			break;
+		case ']':
+			if (editor_prefs.autoclose_chars & GEANY_AC_SBRACKET)
+				isAutoClosed = TRUE;
+			break;
+		case '\'':
+			if (editor_prefs.autoclose_chars & GEANY_AC_SQUOTE)
+				isAutoClosed = TRUE;
+			break;
+		case '"':
```

Comment:
I wonder how Python multiline strings behave with this change?  These ones enclosed by 3 quotoes `""" foo bar """`

Reply:
also when changing `foo("abc")` to `foo(bar("abc"))`